fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java

```diff
@@ -369,4 +375,51 @@ public static long doGetPartitionStatsRowCount(HiveMetaStoreTableInfo hmsTable,
     public static boolean isInternalCatalog(String resourceName) {
         return !resourceName.startsWith("thrift://");
     }
+
+    // Map partition values to partition ranges, eg
+    // [NULL,1992-01-01,1992-01-02,1992-01-03]
+    //               ||
+    //               \/
+    // [0000-01-01, 1992-01-01),[1992-01-01, 1992-01-02),[1992-01-02, 1992-01-03),[1993-01-03, 9999-12-31)
+    public static Map<String, Range<PartitionKey>> getPartitionRange(HiveMetaStoreTable hiveMetaStoreTable,
+                                                                     Column partitionColumn)
+            throws DdlException, AnalysisException {
+        Map<String, Range<PartitionKey>> partitionRangeMap = new LinkedHashMap<>();
+        Map<PartitionKey, Long> partitionKeyMap = hiveMetaStoreTable.getPartitionKeys();
```

Comment:
I'll remove this methods this week. You can use MetadataMgr#listPartitionNames later.